File: csharpcore/StarterBot/Entities/Building.cs
using StarterBot.Enums;

namespace StarterBot.Entities
{
    public class Building : Cell
    {
        public int Health { get; set; }
        public int ConstructionTimeLeft { get; set; }
        public int Price { get; set; }
        public int WeaponDamage { get; set; }
        public int WeaponSpeed { get; set; }
        public int WeaponCooldownTimeLeft { get; set; }
        public int WeaponCooldownPeriod { get; set; }
        public int DestroyScore { get; set; }
        public int EnergyGeneratedPerTurn { get; set; }
        public BuildingType BuildingType { get; set; }
    }
}

File: java/src/main/java/za/co/entelect/challenge/entities/Building.java
package za.co.entelect.challenge.entities;

import za.co.entelect.challenge.enums.BuildingType;
import za.co.entelect.challenge.enums.PlayerType;

public class Building extends Cell {

    public int health;
    public int constructionTimeLeft;
    public int price;
    public int weaponDamage;
    public int weaponSpeed;
    public int weaponCooldownTimeLeft;
    public int weaponCooldownPeriod;
    public int destroyScore;
    public int energyGeneratedPerTurn;
    public BuildingType buildingType;
}
